# Task 3 

### Oppgave 3
#### a) 
We use the expected values that where calculated in 2b. And muplipy them with numberofproduct1 for product 1 and numberofproduct2 for product 


In [1]:
# Oppgave 3

# Results from 2b:
A1 = 0.064157
B1 = 0.102652
C1 = 0.073323
A2 = 0.128314
B2 = 0.128314
C2 = 0.171086

numberofproduct1 = 32
numberofproduct2 = 19

# a) We use the expected values that where calculated in 2b. And multipy them with 32 for product 1 and 19 for product 2. 

B2_time = 0.128314 * numberofproduct2
# New time for machine B product 2: 
new_time_B = B2_time*(1 - 5 / 60)
# We now calculate if there is any overtime and what the cost is:
# Product 1 with reducing:
prod_1_reducing = new_time_B + B1*numberofproduct1 + C1*numberofproduct1
prod_2_reducing = new_time_B + C2*numberofproduct2 + A2*numberofproduct2
# We don't have to pay overtime here. 

# Product 1 without reducing:
prod_1_no_reducing = B2*numberofproduct2 + B1*numberofproduct1 + C1*numberofproduct1
# Product 2 without reducing:
prod_2_no_reducing = B2*numberofproduct2 + C2*numberofproduct2 + A2*numberofproduct2

print(
    f"{prod_1_reducing =}\n",
    f"{prod_2_reducing =}\n",
    f"{prod_1_no_reducing =}\n",
    f"{prod_2_no_reducing =}\n",
)

# The overtime we have to pay is:
Overtime1 = prod_1_no_reducing - 8
Overtime1_cost = Overtime1*100
Overtime2 = prod_2_no_reducing - 8
Overtime2_cost = Overtime2*100

print(f"The overtime cost for product 1: {Overtime1_cost}, product 2: {Overtime2_cost}")
print(f"It pays of to reduce the time, since it won't be any overtime cost")

prod_1_reducing =7.866002166666667
 prod_2_reducing =7.9234021666666665
 prod_1_no_reducing =8.069166
 prod_2_no_reducing =8.126566

The overtime cost for product 1: 6.916599999999917, product 2: 12.65660000000004
It pays of to reduce the time, since it won't be any overtime cost


### Oppgave b
We calcutalte the cost we will get if we 
Since it won't be any overtime we don't have to take that into consideration. 

In [2]:
costfor1 = 10
costfor2 = 12
cost = costfor1*numberofproduct1*0.20 + (costfor1*numberofproduct1+costfor2*numberofproduct2 - (Overtime1_cost + Overtime2_cost))*0.80
print(cost)
print(costfor1*numberofproduct1)
print(costfor1*numberofproduct1+costfor2*numberofproduct2 + Overtime1_cost + Overtime2_cost)
print(f"The profit can decrease with {320/567.57*100} %")

486.74144000000007
320
567.5731999999999
The profit can decrease with 56.38071074933488 %


### Oppgave c
Risk register indentifies potential risks in a project. It keeps track on the likelihood, consequenses of the risk and who is responsible for managing them. 

![riskregister](riskregister.png)

The likelihood of a stock-out proplem is 20%, and therefore we can categorize it as moderate. We calculated earlier that the stock-out could would decrease the proft with 56%, this put the consequense in the Major category. Therefore we can conclude that this risk has extreme inpact on the production. 